src/index.ts

```diff
@@ -1,8 +1,136 @@
+        const pluginInstance = await import(`./app/plugins/${file}`);
+
+        // Try to see if it's in the proper form.
+        try {
+          // Check constructor.
+          const plugin = new pluginInstance.default(container);
+
+          // Check instance.
+          if (!(plugin instanceof Plugin)) {
+            container.loggerService.error(
+              `${file} has a default export, but it is not of type Plugin`
+            );
+            return;
+          }
+
+          // Register plugin.
+          container.pluginService.register(plugin);
+        } catch (err) {
+          container.loggerService.warn(`${file} doesn't have a default export of type Plugin!`);
+        }
+      })
+    );
+
+    // load slash plugins
+    const slashPluginFolder = path.join(__dirname, '/app/slash_plugins');
+    const slashPluginFiles = await readdir(slashPluginFolder);
+    await Promise.allSettled(
+      slashPluginFiles.map(async (file) => {
+        if (!file.endsWith('.ts') && !file.endsWith('.js')) {
+          return;
+        }
+
+        const plugin = await import(`./app/slash_plugins/${file}`).then((m) => m.default);
+
+        const result = SlashCommand.safeParse(plugin);
+        // FIXME we could validate more here: for example, checking the parameters
+        // property, but bot should be fine for now
+        if (result.success) {
+          slashCommands.set(plugin.commandName, result.data as ISlashCommand);
+        } else {
+          container.loggerService.warn(
+            `${file} does not \`export default\` a plugin with type ISlashCommand!`
+          );
+          container.loggerService.warn(result.error);
+        }
+      })
+    );
+
+    const slashCommandUploads = Array.from(slashCommands.entries()).map(([key, command]) => {
+      return {
+        name: key,
+        description: command.description.substring(0, 99),
+        options: command.options,
+      };
+    });
+    // Register commands for all guilds.
+    await Promise.all(
+      container.clientService.guilds.cache.map((guild) => guild.commands.set(slashCommandUploads))
+    );
+
+    // register jobs
+    container.jobService.reset();
+
+    const jobs = container.jobService.jobs;
+    for (const job of jobs) {
+      container.jobService.register(job, container);
+    }
+    // register stores
+    container.storeService.reset();
+
+    container.storeService.stores.forEach((store: Store) => {
+      container.storeService.register(store);
+    });
+    // register webserver
+
+    const defaultPort = 3000;
+    webServer.listen(process.env.WEBSERVER_PORT ?? defaultPort, () =>
+      container.loggerService.info('Webserver is now running')
+    );
+
+    webServer.get('/health', (_, res) => res.send('OK'));
+
+    container.loggerService.info('Bot loaded.');
+    while (true) {
+      const waiting = new Promise((resolve) => setTimeout(resolve, 1_000_000_000));
+      await waiting;
+    }
```

Comment:
Is this still needed even though we're in an `async` context? If so, there probably needs to be some code refactored so it can be `await`ed in this block.

Reply:
Not sure. I didn't touch it because 99% of this PR is just refactoring without any logic changes. The only thing I did remove was some code that purportedly restarted the web server in case of something, but never actually could be called (`91e9bfb`)